dGame/dComponents/PropertyEntranceComponent.cpp

```diff
@@ -71,167 +75,296 @@ void PropertyEntranceComponent::OnEnterProperty(Entity* entity, uint32_t index,
-            entry.Description = description;
+
+        delete friendCheck;
+        friendCheck = nullptr;
+
+        delete friendResult;
+        friendResult = nullptr;
+
+        bool isModeratorApproved = propertyEntry->getBoolean(10);
+
+        if (!isModeratorApproved && entity->GetGMLevel() >= GAME_MASTER_LEVEL_LEAD_MODERATOR) {
+            propertyName = "[AWAITING APPROVAL]";
+            propertyDescription = "[AWAITING APPROVAL]";
+            isModeratorApproved = true;
         }
-        else
-        {
-            entry.Name = "[Awaiting approval] " + name;
-            entry.Description = "[Awaiting approval] " + description;
+
+        bool isAlt = false;
+        // Query to determine whether this property is an alt of the entity.
+        auto isAltQuery = Database::CreatePreppedStmt("SELECT id FROM charinfo where account_id in (SELECT account_id from charinfo WHERE id = ?) AND id = ?;");
+
+        isAltQuery->setInt(1, character->GetID());
+        isAltQuery->setInt(2, owner);
+
+        auto isAltQueryResults = isAltQuery->executeQuery();
+
+        if (isAltQueryResults->next()) {
+            isAlt = true;
         }
-		
-        entry.IsFriend = privacyOption == static_cast<int32_t>(PropertyPrivacyOption::Friends);
-        entry.Reputation = reputation;
-        entry.CloneId = cloneId;
-        entry.IsModeratorApproved = true;
-        entry.AccessType = 3;
 
-        entries.push_back(entry);
+        delete isAltQueryResults;
+        isAltQueryResults = nullptr;
+
+        delete isAltQuery;
+        isAltQuery = nullptr;
+
+        entry = SetPropertyValues(entry, cloneId, ownerName, propertyName, propertyDescription, reputation, isBestFriend, isFriend, isModeratorApproved, isAlt, isOwned, privacyOption, dateLastUpdated, performanceCost);
 
-        delete nameLookup;
+        entries.push_back(entry);
 	}
 
+    delete propertyEntry;
+    propertyEntry = nullptr;
+
     delete propertyLookup;
+    propertyLookup = nullptr;
 
-    /*
-    const auto entriesSize = entries.size();
+    propertyQueries[entity->GetObjectID()] = entries;
 
-    if (startIndex != 0 && entriesSize > startIndex)
-    {
-        for (size_t i = 0; i < startIndex; i++)
-        {
-            entries.erase(entries.begin());
-        }
-    }
-    */
+    uint32_t numberOfProperties = 0;
+    if (sortMethod != SORT_TYPE_FRIENDS || sortMethod != SORT_TYPE_FEATURED) {
+        auto propertiesLeft = Database::CreatePreppedStmt("SELECT COUNT(*) FROM properties WHERE zone_id = ?;");
 
-    propertyQueries[entity->GetObjectID()] = entries;
+        propertiesLeft->setInt(1, this->m_MapID);
+
+        auto result = propertiesLeft->executeQuery();
+        result->next();
+        numberOfProperties = result->getInt(1);
+
+        delete result;
+        result = nullptr;
+
+        delete propertiesLeft;
+        propertiesLeft = nullptr;
+    } else {
+        auto forFriends = query;
+        forFriends.replace(7, 3, "COUNT(*)");
```

Comment:
Maybe it would be good to a comment explaining what this does, or alternatively add a parameter to `BuildQuery` which determines whether it should do `SELECT p.*` or `SELECT COUNT(*)` (I assume), and then call `BuildQuery` here again? This might otherwise be a bit brittle.

(Though feel free to ignore this comment since I am not a member of this project)

Reply:
I agree.  Ill go ahead and also refactor that helper method to make this more robust.

Reply:
This has been updated - please review again if need be!